### PR TITLE
gh-124457: Remove coverity from CPython repo

### PR DESCRIPTION
**Pull request by Mariatta**
Remove coverity from CPython Repo.

There are still some mentions about coverity in the history and changelog for v 2.5. I kept those in there.

* Issue: gh-124457

### CODE REVIEW

File: Misc/NEWS.d/next/Documentation/2024-09-24-11-52-36.gh-issue-124457.yrCjSV.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Remove coverity scan from the CPython repo. It has not been used since 2020
```
Comment (Mariatta) [inline Misc/NEWS.d/next/Documentation/2024-09-24-11-52-36.gh-issue-124457.yrCjSV.rst:1]: Wasn't sure if a Misc News entry is needed, and also I wasn't sure which section should it be.
I chose the "Documentation" section here.

**Comment by vstinner** [comment]
Please keep it. We are using Coverity at Red Hat. cc @stratakis

**Comment by stratakis** [comment]
While we use coverity at Red Hat, it's usually done on specific time frames using [OpenScanHub](https://fedoraproject.org/wiki/OpenScanHub). And we don't utilize the coverity scan model file. So overall I believe the removal is fine. Maybe in the future a nice project would be to introduce coverity scanning through either some service, the buildbots or github actions.

**Comment by vstinner** [comment]
> And we don't utilize the coverity scan model file.

Oh, I didn't know that.

**Review by vstinner — APPROVED** [review]
LGTM. Let's remove it. It's easy to add it again if someone decides to use it again.

**Review by AA-Turner — APPROVED** [review]

**Comment by Mariatta** [comment]
Do I need to backport this?

**Comment by Mariatta** [comment]
And thanks for the reviews!

**Comment by vstinner** [comment]
> Do I need to backport this?

I don't think so.

**Comment by Mariatta** [comment]
Thanks, in this case I'll close the issue.

**Pull request merged**